### ISSUE TITLE
Implement "jackknife-after-bootstrap"

### ISSUE THREAD

**Issue by HDembinski** [dsaxton/resample#50]
The jackknife-after-bootstrap method, as described in Efron and Tibshirani's book, is a clever way to compute an uncertainty for a bootstrap estimate, without computing additional replicas. It needs a bit of additional book-keeping, so it does not come for free, but it is a vast improvement over doing a full jackknife after the bootstrap.

We could add this an keyword option in `resample.bootstrap.bootstrap`, or have a separate `resample.bootstrap.jackknife_after_bootstrap` function. I am leaning slightly towards the latter.

**Comment by dsaxton** [comment]
Yeah that would make a good addition, and I agree it should likely be its own function.

**Comment by HDembinski** [comment]
If we put this functionality in a separate function `resample.bootstrap.jackknife_after_bootstrap`, then we can add this at any time.

However, we also need to consider a possible breaking change in the function `confidence_interval`, which should use this to produce an uncertainty for the interval. A confidence interval without an uncertainty, as we have now, is not really useful.
